### ISSUE TITLE
Meta tags don't match a component variable

### ISSUE THREAD

**Issue by cawa-93** [unjs/unhead#172]
### Environment

Node: 16.20.0
```
-- nuxt@3.6.1
	-- @unhead/vue@1.1.29
```

### Reproduction

https://stackblitz.com/edit/nuxt-starter-dtfjrb?file=app.vue

### Describe the bug

I'm creating a component that should switch the meta tag between three states for `auto`, `dark` and `light` theme mode:

1. ```html
	<meta name="theme-color" content="#121416" media="(prefers-color-scheme: dark)">
	<meta name="theme-color" content="#F8F8F8" media="(prefers-color-scheme: light)">
	```
2. ```html
	<meta name="theme-color" content="#121416" media="screen">
	```

3. ```html
	<meta name="theme-color" content="#F8F8F8" media="screen">
	```

In my reproduction, you can see that the `meta` variable is computed as expected. However, the meta tags from it are not being generated properly.
![зображення](https://github.com/unjs/unhead/assets/1662812/410284c7-c728-4205-9d55-62a522f9e3f6)

Only first render is fine. If you switch `userPreferredMode` `meta` will be recalculated and actual `<meta>` tags will mess up
### Additional context

_No response_

### Logs

_No response_

**Comment by harlan-zw** [comment]
Hey @cawa-93, thanks for the detailed issue.

The issue is related to the hydration from the SSR rendering multiple metatags within the same useHead context. 

The issue is a bit complicated and I'll need to revisit it later, for now, you can work around the issue by wrapping the `useHead` in `process.client` and removing the keys.

See https://stackblitz.com/edit/nuxt-starter-8vxed1?file=app.vue

**Comment by cawa-93** [comment]
Unfortunately, in case with `theme-color` it very important to specify the tags on server and send it to client asap to prevent flash of unstyled bar.

**Comment by P4sca1** [comment]
I also ran into this issue. See https://github.com/nuxt/nuxt/issues/28298 for more context.
A reproduction using the latest nuxt version is available here: https://stackblitz.com/edit/github-ycyewb?file=app.vue.
Basically, the theme-color meta tags are correct during SSR, but after client hydration they get messed up - even while using different keys for each meta tag.

**Comment by P4sca1** [comment]
@harlan-zw Do you need help with fixing this issue? I never worked on this code base, but I can try to debug and fix this issue. Maybe you could provide some starting point on where to look at?

**Comment by harlan-zw** [comment]
This appears to be fixed in v1.11.0-beta.1, see https://stackblitz.com/edit/nuxt-starter-jqlwrq?file=app.vue.
